### PR TITLE
Migration update admitter: Refactor tests

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
- Move the unit tests to the `admitters_test` package in order to explicitly test public API.
- Use fresh admitter instance per test.
- Extract the instantiation of the AdmissionRequest to a single function.
- Expect the full admission response.

Also, simplify the allowed response in the production code.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/migration-update-admitter_test.go [hunk 1]
```diff
@@ -169,7 +224,8 @@ var _ = Describe("Validating MigrationUpdate Admitter", func() {
 
 		admitter := &admitters.MigrationUpdateAdmitter{}
 		resp := admitter.Admit(context.Background(), ar)
-		Expect(resp.Allowed).To(BeTrue())
+
+		Expect(resp).To(Equal(allowedAdmissionResponse()))
```
Comment (Acedus) [inline pkg/virt-api/webhooks/validating-webhook/admitters/migration-update-admitter_test.go:228]: I find it a bit odd that we have `allowedAdmissionResponse` to begin with, consider using `validating_webhooks` `NewPassingAdmissionResponse` as it achieves the exact same.

Comment (orelmisan) [reply pkg/virt-api/webhooks/validating-webhook/admitters/migration-update-admitter_test.go:228]: `allowedAdmissionResponse()` resides in the admitters_test package, and is independent of `validating_webhooks NewPassingAdmissionResponse`.

If we've used the same function twice, the test wasn't independent and would always pass.

Comment (Acedus) [reply pkg/virt-api/webhooks/validating-webhook/admitters/migration-update-admitter_test.go:228]: Ack.

**Comment by iholder101** [comment]
Thank you @orelmisan!
/approve

**Pull request merged**